### ISSUE TITLE
Orientation of controller vs hmd after sync

### ISSUE THREAD

**Issue by gb2111** [cboulay/PSMoveService#353]
After sync position and orientation of hmd and controllers is synced. Orientation is not always same so it can be that if u move in one direction controllers movement is skewed to hmd. You have to try few times with rotation of the controller to get both close.  Can we just set controllers orientation same as where hmd is pointing to avoid the problem?

**Comment by HipsterSloth** [comment]
The problem is that we have two completely independent tracking systems. In order to align them we need a correspondence between a rotation and translation in steamvr tracking space  and a rotation and translation in the PSMove tracking space:

![trackingspaces](https://cloud.githubusercontent.com/assets/16793853/22580466/8836d836-e98d-11e6-9f08-cbba1ce666f9.jpg)

When you press start and select we are correlating the position/rotation of the HMD with the position/rotation of the psmove controller. 

Now all that said, if we KNOW that the PSMove tracking space has the same rotation as the SteamVR tracking space then we could do what you propose. This might be the case with Rift cat tracking setups. Not sure. I could see adding a toggle in the steamvr.settings that sets the alignment mode (rotation/translation or translation only).

**Comment by gb2111** [comment]
could we set same direction of the controller as hmd and leave its rotation?
cause we play with rotation now and perhaps it is not necessary.

**Comment by gb2111** [comment]
I mean if orientation consists of direction and spin then we know that they both have good spin and we need only calibrate direction element of orientation.
Right?

**Comment by gb2111** [comment]
@HipsterSloth 
so basically each time when I run riftcat they are alighed in direction and spin with hmd as you can see (i am looking down directly). all I need is to change position.
I know more or less where this happens however there quite a few computation which is not easy to decode for me. I appreciate you can advise how to make this change.
thanks

**Comment by HipsterSloth** [comment]
I think we could just add a "ignore_alignment_rotation" flag that we check in CPSMoveControllerLatest::FinishRealignHMDTrackingSpace. 

If that flag is true then:

-we set hmd_pose_meters.Orientention to PSMoveQuaternion::identity() here:
https://github.com/cboulay/PSMoveService/blob/master/src/openvr_plugin/driver_psmoveservice.cpp#L2336

And we set controller_pose_meters to PSMoveQuaternion::identity() here:
https://github.com/cboulay/PSMoveService/blob/master/src/openvr_plugin/driver_psmoveservice.cpp#L2385

This would assume that both the PSMove controller and the HMD were pointing down -Z in the SteamVR tracking space.

**Comment by gb2111** [comment]
do u mean to PSMovePose::identity()
https://github.com/cboulay/PSMoveService/blob/master/src/openvr_plugin/driver_psmoveservice.cpp#L2385

as result it is still synced for position and orientation in totally different place.

i would be happy to have orientation ignored for the start. i can try with these code to search for solution hoever if you have further suggestion it would be grt.

**Comment by HipsterSloth** [comment]
No I meant PSMoveQuaternion::identity(). Basically I wanted to remove the rotation from the HMD Pose and the controller pose. That might not be the right thing to do. That alignment code is super confusing even to me. I'll try to take a look tonight and see if I can add that flag. I can't really look at that code right now because I'm at work.

**Comment by HipsterSloth** [comment]
I added a the following new flag in the steamvr.vrsettings:

>    "psmove_settings" : {
      "use_orientation_in_alignment" : false
   },

This will assume that the controller is aligned straight upright facing -Z rather than using the existing controller orientation (which could be slightly off). Let me know if this works for you.

https://github.com/cboulay/PSMoveService/releases/tag/v0.9-alpha6.0.10

**Comment by gb2111** [comment]
@HipsterSloth 
thanks for adding this. but I must say that I was very stupid this time. basically the directions were similat because direction of psmove area is inalignment with direction of game area in my case :) . but it does not mean they are lighted in some way... sorry for this.

so basically I must carefully try original method and go to different part of game arena to see if direction is in synch on the proper level. after some practice it goes very well. the problem is that even small rotation of controler leads to big difference in direction. i think matematically its 
sin(error angle) * distance

i have proposal for alternative synch mechanism to avoid this error but there is no rush to do it.
basically you need capture two points. when first point is taken you can align location of controller and hmd. then you go like one or two meters in other direction and capture point. this gives you precisely the angle. the only assumption here is that you have ability to keep controller in same location on hmd in two points which is my case. so basicall you are sure that in two points hmd and controller are relatively same to each other.

i wonder if I have some time and implement this then would you be able to help me finish 3d calculations? i think it can be bring to actual calculations.
thanks.
